docs/stateful.rst

```diff
@@ -273,6 +273,41 @@ useful sequence of steps will be generated.
 Note that currently preconditions can't access bundles; if you need to use
 preconditions, you should store relevant data on the instance instead.
 
+Invariant
+---------
+
+Often the are invariants that you want to ensure are met after every step in a process.
+It would be possible to add these as rules that are run, but they would be run
+zero or multiple times between other rules. Hypothesis provides a `invariant` decorator,
```

Comment:
In reStructuredText, inline code needs to be surrounded by two backticks, i.e.

```rst
Hypothesis provides a ``invariant`` decorator,
```

I’d also suggest replacing "a" with "an".